doschema/jsvalidation.py

```diff
@@ -0,0 +1,326 @@
+        :param field_type: Type of currently processed field.
+        """
+        self.fields_types_dict[curr_field] = FieldToAdd(
+                schema_index=self.uri,
+                field_tuple=curr_field,
+                field_type=field_type
+            )
+
+    def validate_array(self, field_value, curr_field):
+        """Process each JSON object in the array according to ignore_index.
+
+        :param field_value: Type of the field.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        field_path = curr_field
+        if isinstance(field_value, list):
+            for elem in field_value:
+                if self.ignore_index:
+                    new_field_path = field_path
+                else:
+                    new_field_path = field_path + (field_value.index(elem), )
+
+                self.validate_schema(elem, new_field_path)
+
+        elif isinstance(field_value, dict):
+            self.validate_schema(field_value, curr_field)
+
+    def validate_collection(self, curr_schema, curr_field):
+        """Check schema for allOf, anyOf and oneOf.
+
+        :param curr_schema: Schema or subschema that is currently processed.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        keys = self.COLLECTION_KEYS.intersection(curr_schema)
+        for key in keys:
+            subschema = curr_schema[key]
+
+            if isinstance(subschema, list):
+                for elem in subschema:
+                    self.validate_schema(elem, curr_field)
+
+                props = map(lambda x: x.get('properties'), subschema)
+                flattened_props = {}
+
+                for prop in props:
+                    if prop:
+                        for key, value in prop.items():
+                            flattened_props[key] = value
+
+                for dependency in map(
+                    lambda x: x.get('dependencies'),
+                    subschema
+                ):
+                    if dependency:
+                        self.validate_dependencies(
+                            {'properties': props},
+                            curr_field + ('dependencies', )
+                        )
+            elif isinstance(subschema, dict):
+                self.validate_schema(elem, curr_field)
+
+    def validate_dependencies(self, curr_schema, curr_field):
+        """validate_schema and check dependencies in schema.
+
+        :param curr_schema: Schema or subschema that is currently processed.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        dependencies = curr_schema['dependencies']
+
+        while '$ref' in dependencies:
+            path = dependencies.get('$ref')
+            dependencies = self.resolver.resolve(path)[1]
+
+        for field_name, field_value in dependencies.items():
+            if curr_field + (field_name, ) not in self.fields_types_dict:
```

Comment:
we should at least validate the field_value if it is a `dict`, just in case there is a conflict in subfields.